File: En C/exercice3_5.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
int main ()
{
    char choix , fic[255];
    do
    {  
        printf("entrez un fichier : ");
        scanf ("%s",fic);
        pid_t pid_fils = fork();
        if (pid_fils == 0) execl("/usr/bin/emacs", "emacs", fic , NULL);
        else wait(NULL) ;
        printf(" Entrez Q pour quitter\n");
        scanf (" %c" , &choix) ;  
    }
    while ((choix != 'Q')&&(choix!='q'));
    return(0) ;
}

File: en Python/exercice3_5.py
import os
while True:
    fic = input("entrez un fichier : ")
    pid = os.fork()
    if pid == 0:
        os.execl("/usr/bin/emacs", "emacs", fic)
    os.wait()
    choix = input("Entrez Q pour quitter : ")
    if choix.lower() == 'q':
        break